eppo_core/src/event_ingestion/event_ingestion.rs

```diff
@@ -60,17 +60,28 @@ impl EventIngestionConfig {
 /// A handle to Event Ingestion subsystem.
 pub struct EventIngestion {
     tx: mpsc::Sender<BatchedMessage<Event>>,
+    context_sender: mpsc::Sender<(String, ContextValue)>,
 }
 
 impl EventIngestion {
     /// Starts the event ingestion subsystem on the given background runtime.
     pub fn spawn(runtime: &BackgroundRuntime, config: &EventIngestionConfig) -> EventIngestion {
-        let event_delivery = EventDelivery::new(
+        let event_delivery = Arc::new(Mutex::new(EventDelivery::new(
             reqwest::Client::new(),
             config.sdk_key.clone(),
             config.ingestion_url.clone(),
-        );
-
+        )));
+
+        let event_delivery_clone = Arc::clone(&event_delivery);
+        let (context_sender, mut context_rx) = mpsc::channel::<(String, ContextValue)>(100);
+        runtime.spawn_untracked(async move {
+            while let Some((key, value)) = context_rx.recv().await {
+                let mut event_delivery = event_delivery_clone.lock().await;
+                event_delivery.attach_context(key, value)
+            }
+        });
```

Comment:
minor (not blocking): this mostly works but seems a bit wasteful and may suffer from congestions on event delivery. Given that context is only read/written very briefly, it seems wasteful to lock the whole event delivery with an async mutex (where a sync mutex/rwlock on context would suffice)